src/DataSourcePoller.php

```diff
@@ -105,6 +110,7 @@ public function get_specs_from_data_sources() {
 			$this->read_specs_from_data_sources();
 			$specs = get_transient( $this->args['transient_name'] );
 		}
+		$specs = apply_filters( self::FILTER_NAME_SPECS, $specs, $this->id );
```

Comment:
Allow specs to be filtered before returning them.
I am using this inside the `WcPayPromotion/Init.php` file.